### DISCUSSION TITLE
Trying to compile Alex's 1.34 firmware but get errors but successful compile??

### DISCUSSION THREAD

**Discussion by fgp162** [Jyers/Marlin#1003] [Q&A] [answered]
In Visual Studio Code I get 61 errors and it compiles with success. Here are the errors when compiling the firmware. I have Configuration.h open.
1.) function call is not allowed in a constant expression c/c++
2.) expected an expression c/c++

This is right off github so I wouldn't think there should be this many errors. Can someone possibly tell me what I'm doing wrong? I have installed python and the python extensions for vscode. Know its probably a newbie thing I'm doing wrong. I have tried searching but could not find any answers.

**Comment by fgp162** [comment, marked as answer]
I may have answered my own question. I selected the fault and clicked the light bulb that asked to show possible fixes. When I did it said disable error sqiggles. I did that and now everything compiles without errors with success. Hopefully that is all I needed to do and I didnt just cover up the errors in the software by clicking that.

**Reply by Cal-LF** [reply]
Reference #932 scroll down to Quint097 for solution.
Quote:
I think I might have stumbled on a solution to this problem.

I did the following, which solved the problem for me:

I got to the settings by going to File > Preferences > Setting
I opened the extension settings and scroll down to PlatformIO IDE configuration
I enabled the Use Development PIOCore setting, waited for it to safe and then restarted VS Code.
When VS Code reopened it started reinstalling PlatformIO IDE. I waited for that to finish.
I closed all open folders and then reopened the folder that didn't want to be recognised by PlatformIO.
It now recognised the project correctly and it started generating the correct files in the .vscode folder.
however it didn't solve the problem initially. I had to open one of the files showing errors, click on PlatformIO in the bottom right corner and then click on Edit Configuration (JSON) in the top bar that popped up. This opens .vscode\c_cpp_properties.json but i didn't need to do anything there because when i selected the previous file, the errors were gone.

since making these changes, I haven't had anymore problems or errors.

**Comment by Tuncay-Ayhan** [comment]
You solved it yourself it seems. I get those errors also. I think it’s
because VSC thinks those lines are wrong. Never had any issues with the FW
while ignoring them.

Op ma 14 jun. 2021 om 02:45 schreef fgp162 ***@***.***>
